File: Final-Project-Front/src/app/models/LoanedBook.ts
import {LOAN_STATE} from "./LOAN_STATE";
import {PICKED_UP} from "./PICKED_UP";


export class LoanedBook {


  constructor(
    private _ean: number,
    private _loanState: LOAN_STATE,
    private _loanDate: Date,
    private _returnDate: Date,
    private _pickedUp: PICKED_UP,
    private _userId: number | null,
  ) {
  }


  get ean(): number {
    return this._ean;
  }

  set ean(value: number) {
    this._ean = value;
  }

  get loanState(): LOAN_STATE {
    return this._loanState;
  }

  set loanState(value: LOAN_STATE) {
    this._loanState = value;
  }

  get loanDate(): Date {
    return this._loanDate;
  }

  set loanDate(value: Date) {
    this._loanDate = value;
  }

  get returnDate(): Date {
    return this._returnDate;
  }

  set returnDate(value: Date) {
    this._returnDate = value;
  }


  get pickedUp(): PICKED_UP {
    return this._pickedUp;
  }

  set pickedUp(value: PICKED_UP) {
    this._pickedUp = value;
  }

  get userId(): number | null {
    return this._userId;
  }

  set userId(value: number | null) {
    this._userId = value;
  }


  public toJSON(): any {
    return {
      ean: this._ean,
      loanState: this._loanState,
      loanDate: this._loanDate,
      returnDate: this._returnDate,
      pickedUp: this._pickedUp,
      userId: this._userId,
    };
  }
}

File: Final-Project-Back/edge-service/src/main/java/com/ironhack/edgeservice/controller/dto/LoanedBook.java
package com.ironhack.edgeservice.controller.dto;

import com.ironhack.edgeservice.enums.LOAN_STATE;
import com.ironhack.edgeservice.enums.PICKED_UP;

import java.time.LocalDate;

public class LoanedBook {

    private Long ean;
    private LOAN_STATE LOANSTATE;
    private LocalDate loanDate;
    private LocalDate returnDate;

    private PICKED_UP PICKEDUP;
    private Long userId;

    public LoanedBook() {
    }

    public LoanedBook(Long ean, LOAN_STATE LOANSTATE, LocalDate loanDate, LocalDate returnDate, PICKED_UP PICKEDUP, Long userId) {
        this.ean = ean;
        this.LOANSTATE = LOANSTATE;
        this.loanDate = loanDate;
        this.returnDate = returnDate;
        this.PICKEDUP = PICKEDUP;
        this.userId = userId;
    }


    public Long getEan() {
        return ean;
    }

    public void setEan(Long ean) {
        this.ean = ean;
    }

    public LOAN_STATE getLoanState() {
        return LOANSTATE;
    }

    public void setLoanState(LOAN_STATE LOANSTATE) {
        this.LOANSTATE = LOANSTATE;
    }

    public LocalDate getLoanDate() {
        return loanDate;
    }

    public void setLoanDate(LocalDate loanDate) {
        this.loanDate = loanDate;
    }

    public LocalDate getReturnDate() {
        return returnDate;
    }

    public void setReturnDate(LocalDate returnDate) {
        this.returnDate = returnDate;
    }

    public PICKED_UP getPickedUp() {
        return PICKEDUP;
    }

    public void setPickedUp(PICKED_UP PICKEDUP) {
        this.PICKEDUP = PICKEDUP;
    }

    public Long getUserId() {
        return userId;
    }

    public void setUserId(Long userId) {
        this.userId = userId;
    }
}
